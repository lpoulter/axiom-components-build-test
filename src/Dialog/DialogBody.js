import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Base from '../Base/Base';
import classnames from 'classnames';

export default class DialogBody extends Component {
  static propTypes = {
    /** The main content side a Dialog that occupies most of the space */
    children: PropTypes.node,
    /** Class name to be appended to the element */
    className: PropTypes.string,
  };


  render() {
    const { children, className, ...rest } = this.props;

    const classes = classnames('ax-dialog__body', className);

    return (
      <Base { ...rest } className={ classes }>
        { children }
      </Base>
    );
  }
}
