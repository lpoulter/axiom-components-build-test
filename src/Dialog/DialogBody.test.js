import React from 'react';
import renderer from 'react-test-renderer';
import DialogBody from './DialogBody';

const getComponent = (props = {}) =>
  renderer.create(
    <DialogBody { ...props }>
      Lorem ipsum
    </DialogBody>
  );

describe('DialogBody', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
