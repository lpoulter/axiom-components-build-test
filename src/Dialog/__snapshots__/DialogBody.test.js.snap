// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogBody renders with defaultProps 1`] = `
<div
  className="ax-dialog__body"
>
  Lorem ipsum
</div>
`;
